apps/aecore/src/aec_peer_connection.erl

```diff
@@ -1091,22 +1091,24 @@ send_response(S, Type, Response) ->
     Msg = aec_peer_messages:serialize_response(Type, Response),
     send_msg(S, response, Msg).
 
-%% If the message is more than 65533 bytes it won't fit in a single Noise
-%% message and we need to fragment it.
+%% If the message is more than 65535 - 2 - 16 bytes (2 bytes for the length and
+%% 16 bytes for the encryption MAC) it won't fit in a single Noise message and
+%% we need to fragment it.
 %%
 %% Fragments have the format <<?MSG_FRAGMENT:16, N:16, M:16,
-%% PayLoad:65529/binary>> (where the last fragment has a smaller Payload),
-%% saying that this is fragment N out of M fragments.
+%% PayLoad:65529/binary>> (where the last fragment has a smaller Payload, and
+%% where 16 bytes of the payload is the encryption MAC), saying that this is
+%% fragment N out of M fragments.
 %%
-%% For testing purpose - set the max packet size to 16#FF while testing
+%% For testing purpose - set the max packet size to 16#1FF while testing
 
 -ifndef(TEST).
 -define(MAX_PACKET_SIZE, 16#FFFF).
 -else.
 -define(MAX_PACKET_SIZE, 16#1FF).
 -endif.
--define(FRAGMENT_SIZE, (?MAX_PACKET_SIZE - 6)).
-do_send(ESock, Msg) when byte_size(Msg) < ?MAX_PACKET_SIZE - 2 ->
+-define(FRAGMENT_SIZE, (?MAX_PACKET_SIZE - 6 - 16)).
```

Comment:
It looks to me this shall have `-2` as [2 bytes length added by enoise](https://github.com/aeternity/enoise/blob/71aac15db7b934e0aecac87ff0175abf40aaa2f2/src/enoise_connection.erl#L177).